st2client/st2client/commands/inquiry.py

```diff
@@ -0,0 +1,140 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+
+from st2client.commands import resource
+from st2client.formatters import table
+from st2client.models.inquiry import Inquiry
+
+LOG = logging.getLogger(__name__)
+
+DEFAULT_SCOPE = 'system'
+
+
+class InquiryBranch(resource.ResourceBranch):
+
+    def __init__(self, description, app, subparsers, parent_parser=None):
+        super(InquiryBranch, self).__init__(
+            Inquiry, description, app, subparsers,
+            parent_parser=parent_parser,
+            commands={
+                'list': InquiryListCommand,
+                'get': InquiryGetCommand
+            })
+
+        # Registers extended commands
+        self.commands['respond'] = InquiryRespondCommand(self.resource, self.app,
+                                                         self.subparsers)
+
+        # Remove unsupported commands
+        # TODO: Refactor parent class and make it nicer
+        del self.commands['create']
+        del self.commands['update']
+
+
+class InquiryListCommand(resource.ResourceCommand):
+
+    # Omitting "schema" and "response", as it doesn't really show up in a table well.
+    # The user can drill into a specific Inquiry to get this
+    display_attributes = [
+        'id',
+        'roles',
+        'users',
+        'tag',
+        'ttl'
+    ]
+
+    def __init__(self, resource, *args, **kwargs):
+
+        self.default_limit = 20
+
+        super(InquiryListCommand, self).__init__(
+            resource, 'list', 'Get the list of the %s most recent %s.' %
+            (self.default_limit, resource.get_plural_display_name().lower()),
+            *args, **kwargs)
+
+        self.resource_name = resource.get_plural_display_name().lower()
+        self.parser.add_argument('-n', '--last', type=int, dest='last',
+                                 default=self.default_limit,
+                                 help=('List N most recent %s.' % self.resource_name))
+
+        # Display options
+        self.parser.add_argument('-a', '--attr', nargs='+',
+                                 default=self.display_attributes,
+                                 help=('List of attributes to include in the '
+                                       'output. "all" will return all '
+                                       'attributes.'))
+        self.parser.add_argument('-w', '--width', nargs='+', type=int,
+                                 default=None,
+                                 help=('Set the width of columns in output.'))
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        return self.manager.query_with_count(limit=args.last, **kwargs)
```

Comment:
Can use [query()](https://github.com/StackStorm/st2/blob/master/st2client/st2client/models/core.py#L280)  here, unless we plan to use limit note, eventually.

Reply:
We should use limit note here, like we should in any list api endpoint, imo :)

Reply:
Had to play surgeon on the API side a bit, but finally got the right headers in place, and implemented this table in https://github.com/StackStorm/st2/pull/3653/commits/984fe1f4430930067bdd743594036ead58d705b1